src/sonic-py-common/sonic_py_common/general.py

```diff
@@ -23,3 +24,33 @@ def load_module_from_source(module_name, file_path):
     sys.modules[module_name] = module
 
     return module
+
+
+def getstatusoutput_noshell(cmd):
+    """
+    This function implements getstatusoutput API from subprocess module
+    but using shell=False to prevent shell injection.
+    """
+    p = subprocess.run(cmd, capture_output=True, universal_newlines=True)
+    status, output = p.returncode, p.stdout
+    if output[-1:] == '\n':
+        output = output[:-1]
+    return (status, output)
+
+
+def getstatusoutput_noshell_pipe(cmd1, cmd2):
+    """
+    This function implements getstatusoutput API from subprocess module
+    but using shell=False to prevent shell injection. Input command includes
+    pipe command.
+    """
+    status = 0
+    with subprocess.Popen(cmd1, universal_newlines=True, stdout=subprocess.PIPE) as p1:
+        with subprocess.Popen(cmd2, universal_newlines=True, stdin=p1.stdout, stdout=subprocess.PIPE) as p2:
+            output = p2.communicate()[0]
+            p1.wait()
+            if p1.returncode != 0 and p2.returncode != 0:
+                status = 1
```

Comment:
>status = 1 [](http://example.com/codeflow?start=16&length=10)

This is a general function, so hide information into status 0/1 is not recommended. Suggest return an array of status as the first element in the returned pair.